src/components/NotesPopup/NotesPopup.js

```diff
@@ -54,20 +53,21 @@ class NotesPopup extends Component {
   }
 
   handleAddNoteClicked() {
-    this.props.createNote(this.props.articleId, {
-      text: this.state.currentNote,
-      tmpId: guid(),
-    });
+    this.props.createNote(this.props.articleId, { text: this.state.currentNote });
 
     this.setState({ currentNote: '' });
   }
 
   handleDeleteNote({ noteId, text, noteIndex }) {
-    this.props.deleteNote(this.props.articleId, { noteId, text, noteIndex });
+    this.props.deleteNote(this.props.articleId, noteIndex, { id: noteId, text });
   }
 
-  handleUpdateNote({ noteId, newText, text }) {
-    this.props.updateNote(this.props.articleId, { noteId, newText, text });
+  handleUpdateNote({ noteId, newText, oldText }) {
+    this.props.updateNote(
+      this.props.articleId,
+      { id: noteId, text: newText },
+      { id: noteId, text: oldText }
```

Comment:
`id` should not be included in `newNote`. The server does not support updating ids and might throw an error in the future if we implement more strict validation.